In [5]:
import numpy as np
def calculate_correlation(data1,data2):
    #calculate mean
    mean_data1 = np.mean(data1)
    mean_data2 = np.mean(data2)
    
    #calculate deviations from mean
    dev_data1 = data1 - mean_data1
    dev_data2 = data2 - mean_data2
    
    #calculate cross products of deviations
    cp_data1_data2 = dev_data1 * dev_data2
    
    #calculate correlation coefficients
    corr = np.sum(cp_data1_data2) / np.sqrt(np.sum(dev_data1 ** 2) * np.sum(dev_data2 ** 2))
    
    return corr

#data 
hours_studied = np.array([2, 3, 4, 5, 6, 7])
hours_watching_tv = np.array([4, 3, 2, 1, 0, 0])
outdoor_activity_time = np.array([2, 4, 6, 8, 10, 12])
hours_listening_to_music = np.array([2, 3, 4, 1, 5, 0])
water_consumed = np.array([5, 6, 5, 6, 4, 5])
test_scores = np.array([65, 70, 75, 80, 85, 90])

#print correlation coefficents
print(f"Hours Studied & Test Score:, {calculate_correlation(hours_studied, test_scores):.3f}")
print(f"Hours Watching TV & Test Score:, {calculate_correlation(hours_watching_tv, test_scores):.3f}")
print(f"Outdoor Activity Time & Test Score:, {calculate_correlation(outdoor_activity_time, test_scores):.3f}")
print(f"Hours Listening to Music & Test Score:, {calculate_correlation(hours_listening_to_music, test_scores):.3f}")
print(f"Water Consumed & Test Score:, {calculate_correlation(water_consumed, test_scores):.3f}")

Hours Studied & Test Score:, 1.000
Hours Watching TV & Test Score:, -0.982
Outdoor Activity Time & Test Score:, 1.000
Hours Listening to Music & Test Score:, -0.200
Water Consumed & Test Score:, -0.355


1. Positive Correlation - Hours Studied & Test Score: 1.0 & Outdoor Activity Time & Test Score: 1.0
    Negative Correlation - Hours Watching TV & Test Score:, -0.982 , Hours Listening to Music & Test Score: -0.2 & Water Consumed & Test Score:, -0.355
    
    No significant correlation - Hours Listening to Music & Test Score: -0.2 & Water Consumed & Test Score: -0.355
    
    

2. Hours studied has a high positive correlation with test score because for every one hour increase in studying , test scores get increased by 5 marks. 
This information could help student to target how much they should study to target for a specific test score

3. Hours watching TV has a very high negative correlation with test scores (~0.98).
It means that those who are watching TV a lot has a negative impact on their academics so they cannot manage time properly between leisure and studying

4. Hours listening to music has to significant impact on test scores due to very low negative correlation. 
Still one could say that listening to music a lot could negatively impact test scores


5. Water consumed has a negative correlation with test score which is even though not very high, it's still strange to find how can water consumption negatively impact test score. The reason could be because of a small dataset in which every candidate is drinking sufficient which is not impacting any difference in test scores.

6. The very high positive correlation with Outdoor Activity Time highlights the role of physical activity in maintaining overall well-being, which can positively impact cognitive function and academic success.